### DISCUSSION TITLE
Transitive dependencies

### DISCUSSION THREAD

**Discussion by bever1337** [theory/sqitch#817] [Q&A] [answered]
Suppose my first sqitch change only creates a schema. Later changes will "require" that schema. Now it's been even longer, and my third change requires both previous changes. It is a 'transitive' dependency for the third change to require the schema change.

Does sqitch care about this? Are there ways to positively influence the dependency graph?

In my projects, I am generous with dependencies, sometimes creating transitive ones if the final dependency is very common ex. a schema or shared translation table.

**Comment by theory** [comment, marked as answer]
All the changes in a Sqitch plan inherently depend on the changes that come before. Sqitch won't allow you to deploy a later change without first deploying all the changes before it, in the order listed in the plan file. So dependencies in a single project are largely informational, not enforced.


In that case, do what you're comfortable with! I try to depend only on the changes directly relevant to the current change, not transitive dependencies.

This applies for cross-project dependencies, as well — and here you do need to specify them and Sqitch will enforce them. Require the dependencies required for what's in the current file, but don't worry about the dependencies on that dependency, it should specify them itself.

HTH!

**Reply by bever1337** [reply]
Apologies for accepting this so late. I must have missed the notification. I also think I wrote the original post when I was very sleepy -- it was a denser read than necessary. Thanks for your help and for reading through.
